tests/foreman/ui/test_reporttemplates.py

```diff
@@ -252,17 +253,22 @@ def test_positive_generate_registered_hosts_report(session, module_org, module_l
     """
     # generate Host Status report
     os_name = 'comma,' + gen_string('alpha')
-    os = entities.OperatingSystem(name=os_name).create()
+    os = target_sat.api.OperatingSystem(name=os_name).create()
     host_cnt = 3
     host_templates = [
-        entities.Host(organization=module_org, location=module_location, operatingsystem=os)
+        target_sat.api.Host(organization=module_org, location=module_location, operatingsystem=os)
         for i in range(host_cnt)
     ]
     for host_template in host_templates:
         host_template.create_missing()
-    with session:
+    with target_sat.ui_session() as session:
+        session.organization.select(module_org.name)
+        session.location.select(module_location.name)
         # create multiple hosts to test filtering
-        host_names = [create_fake_host(session, host_template) for host_template in host_templates]
+        host_names = [
+            target_sat.ui_factory(session).create_fake_host(host_template)
```

Comment:
```suggestion
            target_sat.ui_factory.create_fake_host(host_template)
```
no need for the passing extra argument, as we are working on satellite, Satellite will have the `ui` attribute

Reply:
The question is how would your ui attribute have `session` object which is initiated in a test. 

If we create such object in satellite class then it creates a new session object and hence recreates a new session on Grid closes the previoud where the context are set!

Reply:
the same way you are accessing the session here in the first place. So if set the attribute here in this function call `with target_sat.ui_session() as session:` it will be on the Satellite class.